p74/src/bin/rustcrypto_sha3.rs

```diff
@@ -0,0 +1,71 @@
+use sha3::{
+    digest::{Digest, ExtendableOutput, Update, XofReader},
+    Sha3_256, Sha3_512, Shake128, Shake256,
+};
+use std::env;
+use std::fs::File;
+use std::io::{self, Read};
+
+//command line for hash: cargo run --bin rustcrypto_sha3 -- shake256  /Users/olivieradjonyo/Desktop/Rust-Workshop/p74/src/bin/file.txt
+// Generic function to compute the digest for non-XOF algorithms like SHA3-256 and SHA3-512
+fn compute_digest_of_file<T: Digest>(buffer: &[u8]) -> Vec<u8> {
+    let mut hasher = T::new();
+    hasher.update(buffer);
+    hasher.finalize().to_vec()
+}
+//command line for xof: cargo run --bin rustcrypto_sha3 -- shake256 10000  /Users/olivieradjonyo/Desktop/Rust-Workshop/p74/src/bin/file.txt
+// Generic function to compute the XOF for algorithms like SHAKE128 and SHAKE256
+fn compute_xof_of_file<T: Default + Update + ExtendableOutput>(
+    buffer: &[u8],
+    digest_byte_len: usize,
+) -> Vec<u8> {
+    let mut hasher = T::default();
+    hasher.update(buffer);
+    let mut reader = hasher.finalize_xof();
+    let mut result = vec![0u8; digest_byte_len];
+    reader.read(&mut result);
```

Comment:
You can use [`XofRead::read_boxed`](https://docs.rs/digest/latest/digest/trait.XofReader.html#method.read_boxed) here.